sdk/resourcemanager/Azure.ResourceManager.Core/tests/Unit/ResourceTests.cs

```diff
@@ -22,8 +22,8 @@ public class ResourceTests
             "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/testRg/providers/Microsoft.ClassicStorage/storageAccounts/account2")]
         public void CompareToObject(int expected, string id1, string id2)
         {
-            TestResource resource1 = new TestResource(id1);
-            TestResource resource2 = new TestResource(id2);
+            var resource1 = new TestResource<ResourceGroupResourceIdentifier>(id1);
```

Comment:
I think this affects which CompareTo gets called

this method wants to call the one that takes in an TestResource, and the one on line 42 wants to test the one that takes a string

Reply:
Can you clarify? - the only change is that the TestResource type has a generic argument for identifier type.  The Compareto overloads called are the same as they were before the change.